app/diagram/page.tsx

```
@@ -5,9 +5,73 @@ import { CreateDocumentGuard } from "@/components/ui/auth-guard";
 import { Sparkles, Workflow, Zap, Star, Wand2, Share2 } from "lucide-react";
 import { useEffect, useState } from "react";
 import { DiagramGeneratorSkeleton } from "@/components/ui/skeleton";
+import { Button } from "@/components/ui/button";
+import { useToast } from "@/hooks/use-toast";
+import { Button } from "@/components/ui/button";
+import { useToast } from "@/hooks/use-toast";
+
+  const { toast } = useToast();
```

ESLint

[error] React Hook "useToast" cannot be called at the top level. React Hooks must be called in a React function component or a custom React Hook function.

(react-hooks/rules-of-hooks)